src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/Name+Extension.swift

```diff
@@ -8,28 +8,49 @@ import HealthCertificateToolkit
 extension Name {
 
 	var fullName: String {
-		var givenName = self.givenName
-		var familyName = self.familyName
-
-		if givenName?.trimmingCharacters(in: .whitespacesAndNewlines) == "" {
-			givenName = standardizedGivenName
+		
+		let givenName: String?
+		let familyName: String
+		
+		if let value = self.givenName?.trimmingCharacters(in: .whitespacesAndNewlines), !value.isEmpty {
+			givenName = value
+		} else {
+			givenName = readableStandardizedGivenName
 		}
-
-		if familyName?.trimmingCharacters(in: .whitespacesAndNewlines) == "" {
-			familyName = standardizedFamilyName
+		
+		if let value = self.familyName?.trimmingCharacters(in: .whitespacesAndNewlines), !value.isEmpty {
+			familyName = value
+		} else {
+			familyName = readableStandardizedFamilyName
 		}
+		
+		let formatter = PersonNameComponentsFormatter()
+		formatter.style = .long
+
+		var nameComponents = PersonNameComponents()
+		nameComponents.givenName = givenName
+		nameComponents.familyName = familyName
 
-		return [givenName, familyName]
-			.compactMap { $0 }
-			.filter { $0.trimmingCharacters(in: .whitespacesAndNewlines) != "" }
-			.joined(separator: " ")
+		return formatter.string(from: nameComponents)
```

Comment:
Please check the tech spec: https://github.com/corona-warn-app/cwa-app-tech-spec/blob/ae987cda78e5a21b8caec76f1f878ebeecdfaac9/docs/spec/dgc-overview-client.md#determining-the-display-name-of-the-person-from-a-dgc